### DISCUSSION TITLE
How to know a websocket connection is still alive

### DISCUSSION THREAD

**Discussion by maso144** [graphql-dotnet/graphql-client#623] [Q&A]
Hi all

I am facing an issue with a GraphQL subscription on the client side. The subscription stops receiving data from server without any evidence of error. I am struggling to find the root cause of this issue.
There are other clients which receive data from the end point at the same time.

I would like to monitor this connection, for example with ping/pong messages, and re-subscribe if server is not responding.
I could not find any .NET code samples for those ping messages, could you please help?

Or is there a better solution for this issue?

**Comment by Shane32** [comment]
Which GraphQL client library are you using on the client side?

**Reply by maso144** [reply]
I am using these packages:

GraphQL.Client (version 6.0.0)
GraphQL.Client.Abstractions
GraphQL.Client.Abstractions.Websocket
GraphQL.Client.Serializer.Newtonsoft
GraphQL.Primitives

**Comment by Shane32** [comment]
I've moved this discussion to the GraphQL.Client repository.  If you have questions regarding the GraphQL.NET server implementation and its options to prevent timeouts via ping/pong, please post them in the [GraphQL.NET Server repo](https://github.com/graphql-dotnet/server) under issues or discussions, and I can answer there.

**Comment by Shane32** [comment]
I don't use this library, but perhaps @rose-a can help.
